packages/core/src/fhir-to-cda/cda-templates/commons.ts

```diff
@@ -191,21 +217,46 @@ export function formatDateToCdaTimestamp(dateString: string | undefined): string
   if (!dateString) {
     return undefined;
   }
-  const datePart = dateString.replace(/-/g, "");
-  const timePart = "000000";
-  const fractionalSeconds = "0000";
-  const cdaTimeStamp = `${datePart}${timePart}.${fractionalSeconds}`;
-  return cdaTimeStamp;
+  const formatted = dateString.replace(/-/g, "");
+  if (formatted.includes("T")) return formatted.split("T")[0];
+  return formatted;
+}
+
+export function formatDateToHumanReadableFormat(
+  dateString: string | undefined
+): string | undefined {
+  const date = formatDateToCdaTimestamp(dateString);
+  if (!date) return undefined;
+
+  return `${date.slice(0, 4)}-${date.slice(4, 6)}-${date.slice(6, 8)}`;
 }
 
 // see https://build.fhir.org/ig/HL7/CDA-core-sd/StructureDefinition-ST.html
-export function buildValueST(value: string | undefined): CdaValueSt | undefined {
+export function buildValueSt(value: string | undefined): CdaValueSt | undefined {
   if (!value) return undefined;
 
   const valueObject: CdaValueSt = {};
-  valueObject._xsiTypeAttribute = "ST";
-  valueObject._xmlnsXsiAttribute = "http://www.w3.org/2001/XMLSchema-instance";
-  valueObject._inlineTextAttribute = value;
+  valueObject[_xsiTypeAttribute] = "ST";
+  valueObject[_xmlnsXsiAttribute] = "http://www.w3.org/2001/XMLSchema-instance";
+  valueObject[_inlineTextAttribute] = value;
+  return valueObject;
+}
+
+export function buildValueCd(
```

Comment:
can you link the relevant FHIR documentation here?

Reply:
Added a link